File: structural/decorator-example.py
class Person:
    def __init__(self, name, age):
        self.__name = name
        self.__age = age

    def show(self):
        print(self.__name, self.__age)

    # This is default black and white print method
    def print(self):
        pass

    def name(self):
        return self.__name

    def age(self):
        return self.__age


class DecoratedPerson(Person):

    def show(self):
        print(f'Name:{self.name()}, Age:{self.age()}')

    # Here we print colored
    def print(self):
        pass


if __name__ == "__main__":
    p = Person('Alice', 23)
    p.show()

    dp = DecoratedPerson('Bob', 18)
    dp.show()

File: structural/decorator-example.js
class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    show() {
        console.log(this.name, this.age);
    }

    print() { }

    getName() {
        return this.name;
    }

    getAge() {
        return this.age;
    }
}

// ----------------------------------------------------------------------
class DecoratedPerson extends Person {
    show() {
        console.log(`Name:${this.getName()}, Age:${this.getAge()}`);
    }

    print() {
    }
}

// ----------------------------------------------------------------------
let p = new Person('Alice', 23);
p.show();

let dp = new DecoratedPerson('Bob', 18);
dp.show();